src/core/index.ts

```diff
@@ -31,54 +32,66 @@ class WalletSDK {
   }
 
   async receive(offerUri: string): Promise<CredentialResponse> {
-    const client = await Oid4VciClient.fromOfferUrl(offerUri);
-    const credential = await client.getCredential({
-      credential_configuration_id:
-        client.credentialOffer.credential_configuration_ids[0],
-      privateKey: this.jwk,
-    });
+    try {
+      const client = await Oid4VciClient.fromOfferUrl(offerUri);
+      const credential = await client.getCredential({
+        credential_configuration_id:
+          client.credentialOffer.credential_configuration_ids[0],
+        privateKey: this.jwk,
+      });
 
-    return credential;
+      return credential;
+    } catch (e) {
+      throw new DCXException('Failed to receive credential');
```

Comment:
Is it okay not to pass the original error object??

for example
```suggestion
      throw new DCXException(`Failed to receive credential, ${e instanceof Error ? e.message : String(e)}`);
```